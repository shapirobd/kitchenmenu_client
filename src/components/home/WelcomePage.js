import React, { useState, useEffect } from "react";
import axios from "axios";
import { Grid, Typography } from "@material-ui/core";
import { useStyles } from "./styles/WelcomePageStyles";
import RecipeGrid from "../recipes/RecipeGrid";
import IngredientInput from "./IngredientInput";
import IngredientList from "./IngredientList";
import useWindowDimensions from "../../customHooks/getWindowDimensions";

const WelcomePage = () => {
	const classes = useStyles();
	const { width } = useWindowDimensions();

	const [formData, setFormData] = useState("");
	const [ingredients, setIngredients] = useState([]);
	const [formSubmitted, setFormSubmitted] = useState(false);
	const [results, setResults] = useState(undefined);

	const handleChange = (evt) => {
		setFormData(evt.target.value);
	};

	const addIngredient = (evt) => {
		evt.preventDefault();
		setIngredients((ingredients) => [...ingredients, formData]);
	};

	const deleteIngredient = (ingredient) => {
		setIngredients((ingredients) =>
			ingredients.filter((ing) => ing !== ingredient)
		);
	};

	const handleSubmit = (evt) => {
		evt.preventDefault();
		setFormSubmitted(true);
	};

	useEffect(() => {
		if (formSubmitted) {
			const getResults = async () => {
				try {
					const ingredientsParams = ingredients.join(",");
					const recipes = await axios.get(
						"https://api.spoonacular.com/recipes/findByIngredients",
						{
							params: {
								apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
								offset: 0,
								number: 900,
								ingredients: ingredientsParams,
							},
						}
					);
					setResults(recipes.data);
				} catch (e) {
					console.error(e);
				}
			};
			getResults();
		}
	}, [formSubmitted]);

	useEffect(() => {
		if (results) {
			setFormSubmitted(false);
		}
	}, [results]);

	useEffect(() => {
		setFormData("");
	}, [ingredients]);

	return (
		<Grid
			container
			cols={width > 599 ? 2 : 1}
			spacing={1}
			className={width > 599 ? classes.root : classes.mobileRoot}
		>
			{console.log(ingredients)}
			<IngredientInput
				ingredients={ingredients}
				handleChange={handleChange}
				formData={formData}
				addIngredient={addIngredient}
				mobile={width <= 599}
			/>
			<IngredientList
				ingredients={ingredients}
				handleSubmit={handleSubmit}
				deleteIngredient={deleteIngredient}
				mobile={width <= 599}
			/>
			{results ? (
				results.length ? (
					<div
						style={{
							width: "100%",
							backgroundColor: "white",
							display: "flex",
							flexDirection: "column",
							justifyContent: "space-around",
							alignItems: "center",
							borderRadius: "5px",
							margin: "0 0 10px 0",
							paddingBottom: "30px",
						}}
					>
						<Grid
							item
							cols={2}
							xs={12}
							style={{
								margin: "10px 0",
								padding: width > 599 ? "0 10px 0 40px" : 0,
								display: "flex",
								justifyContent: "center",
							}}
						>
							<RecipeGrid feed={results} ingredients={ingredients} />
						</Grid>
					</div>
				) : (
					<div
						style={{
							width: "100%",
							height: "80px",
							display: "flex",
							flexDirection: "column",
							justifyContent: "center",
						}}
					>
						<Typography
							style={{
								color: "red",
								textAlign: "center",
							}}
						>
							No results found
						</Typography>
					</div>
				)
			) : null}
		</Grid>
	);
};

export default WelcomePage;
