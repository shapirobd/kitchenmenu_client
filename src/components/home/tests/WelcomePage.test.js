import React from "react";
import { render, screen } from "@testing-library/react";
import WelcomePage from "../WelcomePage";
import { MemoryRouter } from "react-router-dom";
import { PersistGate } from "redux-persist/integration/react";
import { Provider } from "react-redux";
import { store, persistor } from "../../../configureStore";

it("should render without crashing", () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<WelcomePage />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);
});

it("should match screenshot", () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<WelcomePage />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);

	expect(asFragment()).toMatchSnapshot();
});
