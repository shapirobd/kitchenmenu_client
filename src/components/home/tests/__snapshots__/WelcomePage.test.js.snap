// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-root-20 MuiGrid-container MuiGrid-spacing-xs-1"
    cols="2"
  >
    <div
      class="MuiGrid-root makeStyles-leftGridItem-22 MuiGrid-item MuiGrid-grid-sm-8"
      cols="1"
    >
      <form
        style="width: 100%; height: 100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
      >
        <div
          class="makeStyles-main-24"
        >
          <h4
            class="MuiTypography-root makeStyles-heading-26 MuiTypography-h4"
          >
            What's in your kitchen?
          </h4>
          <div
            class="MuiFormGroup-root MuiFormGroup-row"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-textField-27 MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-29 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-30"
                    style="width: 0.01px;"
                  >
                    <span>
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addItemBtn-28"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add Item
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="MuiGrid-root makeStyles-rightGridItem-35 MuiGrid-item MuiGrid-grid-sm-4"
      cols="1"
    >
      <div
        class="makeStyles-ingredients-37"
      >
        <h6
          class="MuiTypography-root makeStyles-heading-34 MuiTypography-h6"
        >
          Selected Ingredients
        </h6>
        <hr
          class="MuiDivider-root"
          style="width: 100%;"
        />
        <div
          style="height: 87%; width: 100%; background-color: rgba(255, 255, 255, 0.5);"
        >
          <div
            style="display: flex; justify-content: center; flex-wrap: wrap; margin: 10px;"
          />
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-submitBtn-33"
          style="position: absolute; bottom: 0px; width: 100%; height: 10%; box-shadow: -2px 0px 8px rgba(0,0,0,0.5);"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
